#106 - CRUD for users frontend

AmonDeShir opened the pull request:
_This PR will remain a draft until #104 is merged into the main branch._
Should close #106 and close #39.

## PC
![image](https://github.com/user-attachments/assets/a2701fa2-9ab7-438c-8a2e-d5a41791eee5)

## Mobile
![image](https://github.com/user-attachments/assets/a627e625-bc2d-44c1-b664-b16fd85611eb)

EwelinaSkrzypacz reviewed (commented):
Checks are failed

EwelinaSkrzypacz reviewed (commented):
1. Missing validation error about first name in users edit form

![image](https://github.com/user-attachments/assets/34eb6409-19b8-49b2-845d-bb347d81db54)

2. When I edit record and clear all inputs, after click :heavy_check_mark: to save changes, fields are not empty but form is filled with old data :eyes: on edit user form (also on admin edit form)

3. Admin edit form - when I change password for some admin (as a super admin), the password is not changing - admin can still login with old password

AmonDeShir commented:
> 1. Missing validation error about first name in users edit form

fixed

> 2. When I edit record and clear all inputs, after click ✔️ to save changes, fields are not empty but form is filled with old data 👀 on edit user form (also on admin edit form)

fixed

> 3. Admin edit form - when I change password for some admin (as a super admin), the password is not changing - admin can still login with old password

fixed

JokeUrSelf reviewed (commented):
There is no animation of input fieleds when clicking "Edytuj" in AdminPanel
[Screencast from 12-10-2024 02:51:42 PM.webm](https://github.com/user-attachments/assets/7f9f64b3-86e4-40ff-a3d6-5fd2acf564fb)
Same problem in UserPanel, the only diferrence is that the school field is animated, but for some reason very slowly 

The `line-height` of inputs is inconsitent when comparing against other crud pages
Schools
![image](https://github.com/user-attachments/assets/383929f0-8051-476c-ac6d-0a08b9f6171e)
Users
![image](https://github.com/user-attachments/assets/b65ddc4e-8c5e-4e25-aff0-8d43bc7811a4)
Thes second version feels more cluttered

Now that the size of input with `vDynamicInputWidth` depends  on a placeholder, it looks weird when the cursor is being centered
![image](https://github.com/user-attachments/assets/ebb5b881-b6c9-4238-ad07-4113716ae9e4)
Can you make it center only if there is no `value` **and**  `placelolder`

Please add some space between the input field and the list of schools
![image](https://github.com/user-attachments/assets/3a703510-28ce-49aa-a342-06801503bb5b)

Using trash icon for anonymization seems misleading, since it is used to permanently delete an item on other pages
![image](https://github.com/user-attachments/assets/34baff72-6049-41a9-aa05-53571b729aa4)
Consider `UserMinusIcon`
![image](https://github.com/user-attachments/assets/b47be8a2-1dff-4be7-bec1-4561ca0b95ea)

Please use parenthesis when typecasting in template
![image](https://github.com/user-attachments/assets/4cc3482f-2715-424c-9701-c43a465d1a9f)
Because my IDE can't handle the prussure
![image](https://github.com/user-attachments/assets/6f1523f3-7025-4f1e-808a-ddb8d61853de)

AmonDeShir commented on app/Http/Resources/UserResource.php:
```diff
@@ -20,9 +20,12 @@ public function toArray(Request $request): array
             "surname" => $this->surname,
             "email" => $this->email,
             "school" => SchoolResource::make($this->school),
+            "school_id" => $this->school->id,
```
fixed

AmonDeShir commented on resources/js/Types/User.d.ts:
```diff
@@ -4,7 +4,10 @@ interface User {
   surname: string
   email: string
   school: School
+  school_id: number
```
fixed

AmonDeShir commented on resources/js/components/Crud/CrudInput.vue:
```diff
@@ -9,6 +9,7 @@ defineProps<{
   label?: string
```
I don't use it in #title on the UsersPanel and AdminsPanel pages because it automatically takes up 100% of the space, which is problematic when I want to place two inputs (first name, last name) side by side.

AmonDeShir replied:
I fixed QuizHeader and added a format prop to CrudInput.

AmonDeShir commented:
> There is no animation of input fields when clicking "Edytuj" in AdminPanel

fixed

> The `line-height` of inputs is inconsistent when comparing against other crud pages Schools 

fixed
> Now that the size of input with `vDynamicInputWidth` depends on a placeholder, it looks weird when the cursor is being centered 

fixed

> Please add some space between the input field and the list of schools 

fixed

> Using trash icon for anonymization seems misleading, since it is used to permanently delete an item on other pages 

fixed

AmonDeShir commented on resources/js/components/Crud/CrudItem.vue:
```diff
@@ -14,11 +14,12 @@ const props = defineProps<{
   item: T
```
fixed

AmonDeShir commented on resources/js/components/Crud/CrudItem.vue:
```diff
@@ -104,9 +106,11 @@ const showDeleteMessage = ref(false)
             :href="`/admin/${resourceName}/${item.id}`"
             :data="{ ...item as RequestPayload }"
-            @success="editing = false"
+            @success="item = JSON.parse(JSON.stringify(props.item)); editing = false"
```
fixed
